internal/storage/pci/pci.go

```diff
@@ -0,0 +1,84 @@
+package pci
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+// WaitForDevice waits for bus location path of the device to be present
+func WaitForDevice(ctx context.Context, vmbusUuid string) error {
+	_, err := FindDeviceBusLocationFromVmbusUuid(ctx, vmbusUuid)
+	return err
+}
+
+// FindGpuBusLocationFromVmbusUuid finds device bus location by
+// reading /sys/bus/vmbus/devices/<vmbusId>/...
+func FindDeviceBusLocationFromVmbusUuid(ctx context.Context, vmbusUuid string) (string, error) {
+	pciDir, err := findVmbusPciDirs(ctx, vmbusUuid)
+	if err != nil {
+		return "", err
+	}
+
+	pciDeviceLocation, err := findVmbusPciDevice(ctx, pciDir)
+	if err != nil {
+		return "", err
+	}
+	return pciDeviceLocation, nil
+}
+
+// findVmbusPciDevice waits for the pci bus location directory under the path
+// returned from findVmbusPciDirs to exist and returns the full pci bus location or an error
+func findVmbusPciDevice(ctx context.Context, pciDirFullPath string) (string, error) {

```

Comment:
I really like this package! I know it seems a little crazy but you could avoid this double indirection if we make a vmbus package and a pci package here. And in fact that vmbus package would super helpful for other things as well like nic's. 

So you could make vmbus.WaitForDevice(ctx, id string)
Which waits for /sys/bus/vmbus/devices/<id> or timeout. 

Then the pci package is WaitForDevice(ctx, id string) which waits for vmbus.WaitForDevice and then does the /pci* work.

Reply:
I'm good with that change! Could you explain a bit why the vmbus change would be helpful for nic's?

Reply:
Yes. Because this code: https://github.com/microsoft/opengcs/blob/master/internal/network/network.go#L114 is actually just vmbus.WaitForDevice followed by "wait for the net" dir. So we can only have that logic in one place

Reply:
I would still need to wait twice in the pci package, not including the wait that would be in the vmbus package, since I have to wait for a pci* directory to exist under /sys/bus/vmbus/devices/<vmbus guid> and then separately wait for a <pci bus location> directory under that. 

For example, a gpu with a guid of 1111-2222-3333-4444 and pci bus location 1234:00.00.0, we need to wait for:
/sys/bus/vmbus/devices/1111-2222-3333-4444/pci*
and /sys/bus/vmbus/devices/1111-2222-3333-4444/pci1234:00/1234:00*